### ISSUE TITLE
Could you please make the languages selectable from a drop down menu?

### ISSUE THREAD

**Issue by aliquo93** [crow-translate/crow-translate#451]
It would be simpler instead of having to add/remove them manually.

**Comment by Shatur** [comment]
I tried this approach long time ago, languages simply don't fit the screen.

**Comment by aliquo93** [comment]
Because you should not show the selected languages listed in the UI... what I was thinking is a solution like this:

![1](https://user-images.githubusercontent.com/9071297/177617661-f91160ad-d0a4-4c66-9102-fb8d42139585.png)

**Comment by aliquo93** [comment]
> It looks nice, but Qt don't have such widget. It's a GTK feature.



> It looks nice, but Qt don't have such widget. It's a GTK feature.

![2](https://user-images.githubusercontent.com/9071297/177644558-39114957-b019-44c6-897c-293b0548eac8.png)

Well, drop-down menus are possible on Qt too, like Dolphin and Plasma System Monitor in this screen... unfortunately I'm not a programmer so I can't help directly but I hope you would find a solution!

**Comment by aliquo93** [comment]
Or are you meaning that you can't make scrollable drop-down menus? If that's the case, couldn't you make a menu with other sub-menus to not fill the entire screen? Like in this screenshot:

![3](https://user-images.githubusercontent.com/9071297/177645761-015d3521-9496-47aa-ba4b-8af5e288e202.png)

**Comment by Shatur** [comment]
> Or are you meaning that you can't make scrollable drop-down menus?

Yes, I meant scrollable. And with search. Well, it technically possible, but will require writing a custom widget... I do agree that Dialect did it better. Maybe we actually should go this way. Let's reopen the issue. I currently busy at the moment, but maybe someone else will take.

> couldn't you make a menu with other sub-menus to not fill the entire screen?

Will be inconvenient to use, it's 120+ items. It's faster to add a language to panel using search.

**Comment by will-ca** [comment]
In QML/Kirigami I think a modal overlay `ColumnLayout{TextField{}ListView{}}` would be a straightforward and usable way to do it. 

> It looks nice, but Qt don't have such widget.
> 
> It's a GTK feature.

What about something with `QComboBox.setEditable(true)` and `.setCompleter()`? Text boxes with autocompletion/search can definitely be done in QT, which is basically what that feature is.

https://forum.qt.io/topic/41123/solved-search-filter-inside-combobox-qcombobox

https://stackoverflow.com/questions/61508257/using-pyqt5-how-do-i-make-a-qcombobox-searchable

https://doc.qt.io/qt-6/qcombobox.html#completer

https://doc.qt.io/qt-6/qcombobox.html#editable-prop

https://stackoverflow.com/questions/7042430/autocomplete-textbox-in-qt

**Comment by Shatur** [comment]
Feel free to try it out and submit a PR. I will be happy to accept it.
